clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -483,12 +492,7 @@ private synchronized void resetStateAndRejoin() {
         // rebalance in the call to poll below. This ensures that we do not mistakenly attempt
         // to rejoin before the pending rebalance has completed.
         if (joinFuture == null) {
-            // fence off the heartbeat thread explicitly so that it cannot interfere with the join group.
-            // Note that this must come after the call to onJoinPrepare since we must be able to continue
-            // sending heartbeats if that callback takes some time.
-            disableHeartbeatThread();
```

Comment:
We do not need to explicitly disable heartbeat thread since when the state is transited to PREPARING_REBALANCE, the thread would disable itself in the next iteration.